src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -650,8 +642,8 @@ private unsafe RuntimeMethodInfo[] PopulateMethods(Filter filter)
 
                         int numVirtuals = RuntimeTypeHandle.GetNumVirtuals(declaringType);
 
-                        bool* overrides = stackalloc bool[numVirtuals];
-                        new Span<bool>(overrides, numVirtuals).Clear();
+                        Span<bool> overrides = (uint)numVirtuals > 512 ? new bool[numVirtuals] : stackalloc bool[numVirtuals];
```

Comment:
Is 512 here just a heuristic?

Reply:
Yes, like everywhere else? Do you mean it should be a named constant?

Reply:
I've tended to leave a short comment about it being an arbitrary cutoff. I wasn't sure in this particular case if there was meaning to 512 with regards to how many virtuals could exist.

Reply:
Added a comment